lib/api/multiObjectDelete.js

```diff
@@ -241,6 +237,49 @@ function getObjMetadataAndDelete(authInfo, canonicalID, request,
                     }
                     return callback(null, objMD, versionId);
                 }),
+            (objMD, versionId, callback) => {
+                // AWS only returns an object lock error if a version id
+                // is specified, else continue to create a delete marker
+                if (!versionId || !bucket.isObjectLockEnabled()) {
+                    return callback(null, null, objMD, versionId);
+                }
+                const bypassHeader = request.headers['x-amz-bypass-governance-retention'] || '';
+                const hasGovernanceBypass = bypassHeader.toLowerCase() === 'true';
+                if (hasGovernanceBypass && authInfo.isRequesterAnIAMUser()) {
+                    return checkUserGovernanceBypass(request, authInfo, bucket, objMD, log, error => {
+                        if (error) {
```

Comment:
I think we need to be more specific, because the function can return other types of errors (say, maybe if Vault is unavailable or something)

This should work:

```suggestion
                        if (error && error.is.AccessDenied) {
                           // object locked
                        }
                        if (error) {
                           // rest of errors
                        }
```